lib/recorder.js

```diff
@@ -42,19 +43,20 @@ class ConsoleRecorder extends NoOpRecorder {
 }
 
 class HttpRecorder {
-    constructor(url, username, password, debug) {
+    constructor(url, username, password, debug = false, options = {}) {
         function auth(name, pass) {
             const namePass = `${name}:${pass}`;
             return `Basic ${new Buffer(namePass).toString('base64')}`;
         }
 
-        this[HTTP_OPTIONS] = {
-            endpoint: `${url}/hawkular/apm/traces/fragments`,
-            headers: {
-                Authorization: auth(username, password),
-                'Content-Type': 'application/json',
-            },
-        };
+        this[ENDPOINT] = `${url}/hawkular/apm/traces/fragments`;
+        this[HTTP_OPTIONS] = Object.assign({
```

Comment:
I wanted to play with it in browser and it does not work :(. It is still in stage 3 proposal, I will stick with `assign`..